### PR TITLE
CXP-1271: whistelist properties of normalized products

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

In EE, products have metadatas. We don't want those so I added a whitelist of properties.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Persistence/GetProductsQuery.php [hunk 1]
```diff
@@ -114,6 +129,22 @@ private function filterCurrencies(array $products, ?array $currencies = null): a
...
+     * @param array<Product> $products
+     * @param array<string> $whitelist
+     * @return array<Product>
+     */
+    private function filterNormalizedProperties(array $products, array $whitelist): array
+    {
+        $keys = array_flip($whitelist);
+
+        foreach ($products as &$product) {
+            $product = array_intersect_key($product, $keys);
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Persistence/GetProductsQuery.php:142]:
```suggestion
        $keys = \array_flip($whitelist);

        foreach ($products as &$product) {
            $product = \array_intersect_key($product, $keys);
```

**Review by masacc — APPROVED** [review]

**Review by willy-ahva — APPROVED** [review]
Could you add a test?

**Comment by tseho** [comment]
> Could you add a test?

Which test do you have in mind ?
It only break in EE conditions.

**Comment by willy-ahva** [comment]
> > Could you add a test?
> 
> Which test do you have in mind ? It only break in EE conditions.

Maybe a test that checks the white list is well applied

**Comment by tseho** [comment]
> Maybe a test that checks the white list is well applied

I understood that part, but how ? the class is already covered by integration tests. So we know that it didn't break the expected behavior.
Do you suggest to write unit tests on the class that will mock both GetConnectorProduct & its normalizer ?

**Comment by tseho** [comment]
I'm merging because we need to unlock the CI, I will gladly continue this discussion and add tests if necessary.

**Pull request merged**